forge-game/src/main/java/forge/game/card/CardState.java

```diff
@@ -452,6 +461,22 @@ public FCollectionView<ReplacementEffect> getReplacementEffects() {
             }
             result.add(loyaltyRep);
         }
+        if (getTypeWithChanges().isBattle()) {
```

Comment:
maybe getTypeWithChanges just once before the checks so it doesn't get called multiple times if isn't needed?